src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagCache.cs

```diff
@@ -0,0 +1,64 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class CustomTagCache : ICustomTagCache
```

Comment:
comment on the class responsibility

Reply:
Also do you really want to implement cache now? We should really look at using .net Memory cache or something like that, this is not thread safe. We will need a lot of advanced telemetry and ops to control the expiration and size.

Reply:
* CustomTagCache is foundation of the entire optimization. QIDO and STOW depends on this.
* Yes, it's not thread safe, will figure out a way to handle with
* Could you details bit more on "need a lot of advanced telemetry and ops"?

Reply:
details on  "need a lot of advanced telemetry and ops":
* for Metrix, need to consider miss and hit rate
* check https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-5.0 to see what to consider when design cache.